api/internal/plugins/compiler/compiler_test.go

```diff
@@ -18,34 +23,32 @@ func TestCompiler(t *testing.T) {
 	if err != nil {
 		t.Error(err)
 	}
-	c := NewCompiler(srcRoot)
-
-	c.SetGVK("someteam.example.com", "v1", "DatePrefixer")
-	expectObj := filepath.Join(
-		srcRoot, "someteam.example.com", "v1", "dateprefixer", "DatePrefixer.so")
-	if expectObj != c.ObjPath() {
-		t.Errorf("Expected '%s', got '%s'", expectObj, c.ObjPath())
-	}
-	err = c.Compile()
-	if err != nil {
-		t.Error(err)
-	}
-	if !utils.FileExists(expectObj) {
-		t.Errorf("didn't find expected obj file %s", expectObj)
+	for i, obj := range srcRoot {
```

Comment:
why `obj`? It's a possible root dir, no? In any case, I think that this is another spot where the srcRoot refers to part of Kustomize's test setup, so we should be able to assert that the length is one and then proceed without a loop.

Reply:
This is another area where @monopole may need to correct me, but it seems to me that `DeterminePluginSrcRoot` is always used for checking Kustomize test plugin roots, in which case it could be modified to continue to return a single value / return an error if more than one value was found. That way the test cases won't all have to know about that.

Reply:
Shouldn't then, we modify `DeterminePluginSrcRoot` to return a (string and error)?